docs/logicapps-ai-course/04_agent_functionality/05-extend-knowledge-to-agent-using-ai-search-integrated-vectorization.md

```diff
@@ -0,0 +1,72 @@
+> - You have access to an Azure AI Search service. For more on creating this resource visit here [Create an Azure AI Search service](https://learn.microsoft.com/en-us/azure/search/tutorial-optimize-indexing-push-api#create-an-azure-ai-search-service). Additionally, this module assumes your search index is created using this index schema: [index_schema](media/05-extend-knowledge-to-agent-using-ai-search-integrated-vectorization/integrated_vectorizer_schema.json)
+>  - Be sure to supply your own OpenAI resource values for the following: {OpenAI resource URI}, {deployment ID}, {API key}, and { model name }.
+
+### Part 1 - Create our data ingestion workflow
+Refer to [Module 4](https://azure.github.io/logicapps-labs/docs/logicapps-ai-course/agent_functionality/extend-knowledge-to-agent#step-1---create-our-data-ingestion-worflow) - Step 1 for creating your ingestion workflow.
+
+### Part 2 - Create your knowledge agent
+To simplify the retrieval workflow compared to module 4, you will use the **Azure AI Search natural language** action in your workflow. This action will automatically engage the integrated vectorizer for generating embeddings. Because the search service creates embeddings for the incoming user queries, your Logic App will not need a separate action to generate embeddings.
+
+#### Benefits
+- Fewer actions in your Logic App (simpler workflows)
+- Reduced API calls and maintenance
+- Consistent vectorization when the same model is used for indexing and querying
+
+#### Step 1 - Create your retrieval workflow
+1. In the [Azure portal](https://portal.azure.com), open your Standard logic app resource.
+
+1. Find and open your conversational agent workflow in the designer.
+
+
+On the designer, select the agent action. Rename the agent: **Document knowledge agent**. Next enter the System Instructions  
+```
+You are a helpful assistant, answering questions about specific documents. When a question is asked, follow these steps in order: 
+
+Use this tool to do a vector search of the user's question, the output of the vector search tool will have the related information to answer the question. Use the "content" field to generate an answer. Use only information to answer the user's question. No other data or information should be used to answer the question.
+```
+1. On the designer, inside the agent, select the plus sign (+) under **Add tool**.
+1. Click on the Tool, and rename it to **Document search tool**. Then add the follow Description **Searches an azure search index for content related to the input question.**
+1. Click the plus **(+)** sign to add a new action.
+1. Search for **Azure AI Search (built-in)**.
+1. Select the **Search vectors with natural language** action.
+![Screenshot of Azure AI Search available actions.](media/05-extend-knowledge-to-agent-using-ai-search-integrated-vectorization/integrated_search.png)
```

Comment:
you may want to highlight the action that you want the reader to select in the image